src/test/features/response/routes/when-did-you-pay.ts

```diff
@@ -0,0 +1,115 @@
+  attachDefaultHooks()
+
+  describe('on GET', () => {
+    const method = 'get'
+    checkAuthorizationGuards(app, method, pagePath)
+    checkNotDefendantInCaseGuard(app, method, pagePath)
+
+    context('when user authorised', () => {
+      beforeEach(() => {
+        idamServiceMock.resolveRetrieveUserFor(claimStoreServiceMock.sampleClaimObj.defendantId, 'citizen')
+      })
+
+      checkAlreadySubmittedGuard(app, method, pagePath)
+
+      context('when response not submitted', () => {
+        it('should return 500 and render error page when cannot retrieve claim', async () => {
+          claimStoreServiceMock.rejectRetrieveClaimByExternalId('HTTP error')
+
+          await request(app)
+            .get(pagePath)
+            .set('Cookie', `${cookieName}=ABC`)
+            .expect(res => expect(res).to.be.serverError.withText('Error'))
+        })
+
+        it('should render page when everything is fine', async () => {
+          draftStoreServiceMock.resolveFind('response')
+          claimStoreServiceMock.resolveRetrieveClaimByExternalId()
+          await request(app)
+            .get(pagePath)
+            .set('Cookie', `${cookieName}=ABC`)
+            .expect(res => expect(res).to.be.successful.withText('When did you pay?'))
+        })
+      })
+    })
+  })
+
+  describe('on POST', () => {
+    const method = 'post'
+    checkAuthorizationGuards(app, method, pagePath)
+    checkNotDefendantInCaseGuard(app, method, pagePath)
+
+    context('when user authorised', () => {
+      beforeEach(() => {
+        idamServiceMock.resolveRetrieveUserFor(claimStoreServiceMock.sampleClaimObj.defendantId, 'citizen')
+      })
+
+      checkAlreadySubmittedGuard(app, method, pagePath)
+
+      context('when response not submitted', () => {
+        context('when form is invalid', () => {
+          it('should return 500 and render error page when cannot retrieve claim', async () => {
+            claimStoreServiceMock.rejectRetrieveClaimByExternalId('HTTP error')
+
+            await request(app)
+              .post(pagePath)
+              .set('Cookie', `${cookieName}=ABC`)
+              .expect(res => expect(res).to.be.serverError.withText('Error'))
+          })
+        })
+
+        context('when form is valid', () => {
+          it('should return 500 and render error page when form is valid and cannot save draft', async () => {
+            draftStoreServiceMock.resolveFind('response')
+            draftStoreServiceMock.rejectSave()
+            claimStoreServiceMock.resolveRetrieveClaimByExternalId()
+
+            await request(app)
+              .post(pagePath)
+              .set('Cookie', `${cookieName}=ABC`)
+              .send({ date: { year: '1978', month: '1', day: '11' }, text: 'Paid cash' })
+              .expect(res => expect(res).to.be.serverError.withText('Error'))
+          })
+
+          it('should redirect to timeline page when form is valid and everything is fine', async () => {
```

Comment:
Please correct test case description as there is no redirect to timeline, there is redirect to task list.